service/history/workflow/cache/cache.go

```diff
@@ -190,7 +197,7 @@ func (c *CacheImpl) getOrCreateWorkflowExecutionInternal(
 	if !cacheHit {
 		handler.Counter(metrics.CacheMissCounter.GetMetricName()).Record(1)
 		// Let's create the workflow execution workflowCtx
-		workflowCtx = workflow.NewContext(c.shard, key, c.logger)
+		workflowCtx = workflow.NewContext(shardContext, key, c.logger)
```

Comment:
when shards are moving, we can have two shard context for the same shardID on the host. And the new shard may get a workflowCtx that references the old shard context I think?

Reply:
Also right now when shard reload, each shard has a clean/empty cache, and will load workflows from DB. Moving to host level, a new shard context can potentially load mutable state from old shard context. and I am not sure if this will cause any issue.

The safest approach I can think of is make shard context identity (e.g. shard context's initial rangeID?) part of the cache key, which makes sure shard contexts are isolated.

Reply:
Change to not caching shard context.